src/main/java/monitoringsystemturbo/presenter/ExportPresenter.java

```diff
@@ -1,107 +1,178 @@
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.CheckBox;
+import javafx.stage.Stage;
+import monitoringsystemturbo.controller.ConfirmExportController;
+import monitoringsystemturbo.exporter.MainExporter;
+import monitoringsystemturbo.model.TrackingService;
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+
+import static java.time.LocalDate.now;
+
+public class ExportPresenter {
+
+    private Stage primaryStage;
+    private MainExporter mainExporter;
+    private TrackingService trackingService;
+
+    @FXML
+    private JFXDatePicker fromDatePicker;
+
+    @FXML
+    private JFXDatePicker toDatePicker;
+
+    @FXML
+    private JFXTimePicker toTimePicker;
+
+    @FXML
+    private JFXTimePicker fromTimePicker;
+
+    @FXML
+    private CheckBox wholeRangeCheckBox;
+
+    @FXML
+    private CheckBox fromBeginCheckBox;
+
+    @FXML
+    private CheckBox toNowCheckBox;
+
+    @FXML
+    public void initialize() {
+        fromDatePicker.setValue(now());
+        toDatePicker.setValue(now());
+        fromTimePicker.setValue(LocalTime.of(0, 0));
+        toTimePicker.setValue(LocalTime.now());
+    }
+
+    @FXML
+    public void onConfirm() {
+        LocalDateTime fromTime = fromDatePicker.getValue().atTime(fromTimePicker.getValue().getHour(), fromTimePicker.getValue().getMinute());
+        LocalDateTime toTime = toDatePicker.getValue().atTime(toTimePicker.getValue().getHour(), toTimePicker.getValue().getMinute());
+
+        Date dateStart = Date.from(fromTime.atZone(ZoneId.systemDefault()).toInstant());
+        Date dateEnd = Date.from(toTime.atZone(ZoneId.systemDefault()).toInstant());
+
+        if (dateStart.after(dateEnd)) {
+            Alert errorAlert = new Alert(Alert.AlertType.ERROR);
+            errorAlert.setTitle("Error!");
+            errorAlert.setHeaderText(null);
+            errorAlert.setContentText("Error occurred while checking date.");
+            errorAlert.showAndWait();
+        } else {
+            ConfirmExportController confirmExportController = new ConfirmExportController(primaryStage);
+            try {
+                List<String> applicationsToExport = confirmExportController.showConfirmationAndGetAppList(trackingService.getApplicationsNames());
+                if (!confirmExportController.getCancelValue()) {
+                    if (wholeRangeCheckBox.isSelected()) {
+                        mainExporter.export(trackingService, applicationsToExport);
+                    } else {
+                        mainExporter.export(trackingService, applicationsToExport, fromTime, toTime);
+                    }
+                    Alert successAlert = new Alert(Alert.AlertType.INFORMATION);
+                    successAlert.setTitle("Success!");
+                    successAlert.setHeaderText(null);
```

Comment:
especially you can make a method for alert displaying, it almost the same code all over the place just with different arguments